src/components/network-area-diagram-viewer/diagram-utils.ts

```diff
@@ -436,3 +436,48 @@ export function getBoundarySemicircle(
         getCirclePath(busOuterRadius, startAngle, startAngle + Math.PI, true)
     );
 }
+
+// get egde name elements: position and angle elements
+export function getEdgeNameElements(
+    edgeNode: SVGGraphicsElement
+): [SVGGraphicsElement | null, SVGGraphicsElement | null] {
+    let edgeNamePositionElement: SVGGraphicsElement | null = null;
+    let edgeNameAngleElement: SVGGraphicsElement | null = null;
+    // get DOM element containing name
+    const nameElement: SVGGraphicsElement =
+        edgeNode.lastElementChild as SVGGraphicsElement;
+    if (
+        nameElement != null &&
+        nameElement.tagName == 'g' &&
+        (nameElement.id == null || nameElement.id.length == 0)
+    ) {
+        // get DOM element containing name position
+        const namePositionElement: SVGGraphicsElement =
+            nameElement.lastElementChild as SVGGraphicsElement;
+        if (
+            namePositionElement != null &&
+            namePositionElement.tagName == 'g' &&
+            (namePositionElement.id == null ||
+                namePositionElement.id.length == 0)
+        ) {
+            edgeNamePositionElement = namePositionElement;
+            // get DOM element containing name angle
+            const nameAngleElement: SVGGraphicsElement =
+                namePositionElement.lastElementChild as SVGGraphicsElement;
+            if (
+                nameAngleElement != null &&
+                nameAngleElement.tagName == 'text'
+            ) {
+                edgeNameAngleElement = nameAngleElement;
+            }
```

Comment:
similarly
```ts
const edgeNameAngleElement: SVGGraphicsElement =
        edgeNamePositionElement.querySelector("text") as SVGGraphicsElement;
```

Reply:
Used querySelector for getting the element